rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1850,6 +1913,135 @@ public Response deleteStudentContactLogEntry(@PathParam("STUDENTID") Long studen
+      return Response.status(Status.BAD_REQUEST).build();
+    }
+
+    Student student = studentController.findStudentById(studentId);
+    Status studentStatus = checkStudent(student);
+    if (studentStatus != Status.OK)
+      return Response.status(studentStatus).build();
+    
+    StudentContactLogEntryComment contactLogEntryComment = studentContactLogEntryCommentController.findContactLogEntryCommentById(commentId);
+    if (contactLogEntryComment == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+
+    StudentContactLogEntry contactLogEntry = contactLogEntryComment.getEntry();
+
+    if (contactLogEntry == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    if (contactLogEntry.getArchived()) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+
+    if (!contactLogEntry.getStudent().getId().equals(contactLogEntry.getStudent().getId())) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    User loggedUser = sessionController.getUser();
+    StaffMember staffMember = userController.findStaffMemberById(loggedUser.getId());
+    
+    ContactLogAccess access = studentController.resolveContactLogAccess(student);
+    
+    if (access.equals(ContactLogAccess.ALL)) {
+      return Response.ok(objectFactory.createModel(studentContactLogEntryCommentController.updateContactLogEntryComment(contactLogEntryComment, entity.getText(), entity.getCommentDate()))).build();
+    } else if (access.equals(ContactLogAccess.OWN)) {
+      if (contactLogEntryComment.getCreator() != null) {
+        if (contactLogEntryComment.getCreator().getId().equals(staffMember.getId())) {
+          return Response.ok(objectFactory.createModel(studentContactLogEntryCommentController.updateContactLogEntryComment(contactLogEntryComment, entity.getText(), entity.getCommentDate()))).build();
+        }
+      }
+    } 
+    
+    return Response.status(Status.FORBIDDEN).build();
+  }
+
+  @Path("/students/{STUDENTID:[0-9]*}/contactLogEntries/entryComments/{COMMENTID:[0-9]*}")
+  @DELETE
+  @RESTPermit(handling = Handling.INLINE)
+  public Response deleteStudentContactLogEntryComment(@PathParam("STUDENTID") Long studentId, 
+      @PathParam("COMMENTID") Long commentId,
+      @DefaultValue("false") @QueryParam("permanent") Boolean permanent) {
+    Student student = studentController.findStudentById(studentId);
+    Status studentStatus = checkStudent(student);
+    if (studentStatus != Status.OK)
+      return Response.status(studentStatus).build();
+
+    StudentContactLogEntryComment contactLogEntryComment = studentContactLogEntryCommentController.findContactLogEntryCommentById(commentId);
+    if (contactLogEntryComment == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    StudentContactLogEntry contactLogEntry = contactLogEntryComment.getEntry();
+    if (!contactLogEntry.getStudent().getId().equals(contactLogEntry.getStudent().getId())) {
```

Comment:
Am I blind or is this comparing `contactLogEntry.getStudent().getId()` to itself?